lib/Test/WWW/Stub/Handler.pm

```diff
@@ -0,0 +1,71 @@
+package Test::WWW::Stub::Handler;
+
+use strict;
+use warnings;
+
+use Carp qw(confess);
+
+sub new {
+    my ($class, %args) = @_;
+    my $pattern = $args{pattern};
+    my $app     = $args{app};
+    return bless { pattern => $pattern, app => $app }, $class;
+}
+
+sub factory {
+    my ($class, $uri_or_re, $app_or_res) = @_;
+    my $app;
+    my $type = ref($app_or_res);
+    if ($type eq 'CODE') {
+        $app = $app_or_res;
+    } elsif ($type eq 'ARRAY') {
+        $app = sub { $app_or_res };
+    } else {
+        confess 'Handler MUST be a PSGI app or an ARRAY';
+    }
+    return $class->new(pattern => $uri_or_re, app => $app);
+}
+
+sub is_regexp {
+    my ($self) = @_;
+    return ref($self->{pattern}) eq 'Regexp';
+}
+
+sub is_str {
```

Comment:
evaluating `is_str` against an instance of `URI.pm` returns true, but IMO it should return false.
Though current behavior of `_match` should be kept unchanged (We want to match an instance of `URI.pm` by string comparison), but this method name `is_str` is misleading.

```perl
use strict;
use warnings;
use Test::More;

use URI;
use Test::WWW::Stub::Handler;

my $handler = Test::WWW::Stub::Handler->new(
    pattern => URI->new('http://www.google.com/'),
    app => sub { [] },
);

ok !$handler->is_regexp;
{
    local $TODO = 'pattern is an instance of URI, so `is_str` should not be true';
    ok !$handler->is_str;
}

done_testing;

__END__
$ carton exec -- prove -lvr t/20_handler.t
t/20_handler.t ..
ok 1
not ok 2 # TODO pattern is an instance of URI, so `is_str` should not be true

#   Failed (TODO) test at t/20_handler.t line 16.
1..2
ok
All tests successful.
Files=1, Tests=2,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.02 cusr  0.00 csys =  0.04 CPU)
Result: PASS
```